test/test_dot_device.cc

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2017-2022, University of Tennessee. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#include "test.hh"
+#include "cblas_wrappers.hh"
+#include "lapack_wrappers.hh"
+#include "blas/flops.hh"
+#include "print_matrix.hh"
+
+// -----------------------------------------------------------------------------
+template< typename Tx, typename Ty >
+void test_dot_device_work( Params& params, bool run )
+{
+    using namespace testsweeper;
+    using namespace blas;
+    typedef scalar_type<Tx, Ty> scalar_t;
+    typedef real_type<scalar_t> real_t;
+    typedef long long lld;
+
+    // get & mark input values
+    char mode       = params.pointer_mode();
+    int64_t n       = params.dim.n();
+    int64_t incx    = params.incx();
+    int64_t incy    = params.incy();
+    int64_t device  = params.device();
+    int64_t verbose = params.verbose();
+
+    scalar_t  result_host;
+    scalar_t* result = &result_host;
+    scalar_t  result_cblas;
+
+    // mark non-standard output values
+    params.gflops();
+    params.gbytes();
+    params.ref_time();
+    params.ref_gflops();
+    params.ref_gbytes();
+
+    // adjust header to msec
+    params.time.name( "BLAS++\ntime (ms)" );
+    params.ref_time.name( "Ref.\ntime (ms)" );
+
+    if (! run)
+        return;
+
+    if (blas::get_device_count() == 0) {
+        params.msg() = "skipping: no GPU devices or no GPU support";
+        return;
+    }
+
+    // setup
+    size_t size_x = (n - 1) * std::abs(incx) + 1;
+    size_t size_y = (n - 1) * std::abs(incy) + 1;
+    Tx* x    = new Tx[ size_x ];
+    Tx* xref = new Tx[ size_x ];
+    Ty* y    = new Ty[ size_y ];
+    Ty* yref = new Ty[ size_y ];
+
+    // device specifics
+    blas::Queue queue(device, 0);
```

Comment:
In BLAS++ and LAPACK++ the preferred convention is to have spaces inside function call parens:
`queue( device, 0 )`. Here and function calls below.

SLATE originally had basically the opposite convention, but I think we're moving towards adding spaces for better readability.